druid/src/data.rs

```diff
@@ -375,6 +387,58 @@ impl Data for piet::Color {
     }
 }
 
+//FIXME Vector::ptr_eq is not currently reliable; this is a temporary impl?
+#[cfg(feature = "im")]
+impl<T: Data> Data for im::Vector<T> {
+    fn same(&self, other: &Self) -> bool {
+        // for reasons outlined in https://github.com/bodil/im-rs/issues/129,
+        // ptr_eq always returns false for small collections. This heuristic
+        // falls back to using equality for collections below some threshold.
+        // There may be a possibility of this returning false negatives, but
+        // not false positives; that's an acceptable outcome.
+
+        /* this is the impl I expected to use
+        const INLINE_LEN: usize = 48; // bytes available before first allocation;
+        let inline_capacity: usize = INLINE_LEN / std::mem::size_of::<T>();
+        if self.len() <= inline_capacity {
+            self.iter().zip(other.iter()).all(|(a, b)| a.same(b))
+        } else {
+            self.ptr_eq(other)
+        }
+        */
+
+        self.iter().zip(other.iter()).all(|(a, b)| a.same(b))
```

Comment:
This isn't correct after all, because it will stop comparing after the shorter iterator. So this will return `true` for `[1, 2, 3]` vs `[1, 2, 3, 4]`.

The correct way would be to use [`eq_by`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.eq_by) but that's nightly only.

So something like the following is needed:

```rust
self.len() == other.len() && self.iter().zip(other.iter()).all(|(a, b)| a.same(b))
```

Reply:
oops, good catch